x-pack/plugins/cases/server/authorization/authorization.ts

```diff
@@ -79,70 +79,106 @@ export class Authorization {
-
-    if (securityAuth && this.shouldCheckAuthorization()) {
-      const requiredPrivileges: string[] = [securityAuth.actions.cases.get(owner, operation.name)];
-
-      const checkPrivileges = securityAuth.checkPrivilegesDynamicallyWithRequest(this.request);
-      const { hasAllRequested, username } = await checkPrivileges({
-        kibana: requiredPrivileges,
-      });
+  public async ensureAuthorized({
+    owner,
+    operation,
+    savedObjectID,
+  }: {
+    owner: string;
+    operation: OperationDetails;
+    savedObjectID: string;
+  }) {
+    try {
+      const { securityAuth } = this;
+      const isOwnerAvailable = this.featureCaseOwners.has(owner);
+
+      if (securityAuth && this.shouldCheckAuthorization()) {
+        const requiredPrivileges: string[] = [
+          securityAuth.actions.cases.get(owner, operation.name),
+        ];
+
+        const checkPrivileges = securityAuth.checkPrivilegesDynamicallyWithRequest(this.request);
+        const { hasAllRequested, username } = await checkPrivileges({
+          kibana: requiredPrivileges,
+        });
+
+        if (!isOwnerAvailable) {
+          /**
+           * Under most circumstances this would have been caught by `checkPrivileges` as
+           * a user can't have Privileges to an unknown owner, but super users
+           * don't actually get "privilege checked" so the made up owner *will* return
+           * as Privileged.
+           * This check will ensure we don't accidentally let these through
+           */
+          throw Boom.forbidden(this.authLogger.failure({ username, owner, operation }));
+        }
 
-      if (!isOwnerAvailable) {
-        /**
-         * Under most circumstances this would have been caught by `checkPrivileges` as
-         * a user can't have Privileges to an unknown owner, but super users
-         * don't actually get "privilege checked" so the made up owner *will* return
-         * as Privileged.
-         * This check will ensure we don't accidentally let these through
-         */
-        throw Boom.forbidden(this.auditLogger.failure({ username, owner, operation }));
+        if (hasAllRequested) {
+          this.authLogger.success({ username, operation, owner });
+        } else {
+          throw Boom.forbidden(this.authLogger.failure({ owner, operation, username }));
+        }
+      } else if (!isOwnerAvailable) {
+        throw Boom.forbidden(this.authLogger.failure({ owner, operation }));
       }
 
-      if (hasAllRequested) {
-        this.auditLogger.success({ username, operation, owner });
-      } else {
-        throw Boom.forbidden(this.auditLogger.failure({ owner, operation, username }));
-      }
-    } else if (!isOwnerAvailable) {
-      throw Boom.forbidden(this.auditLogger.failure({ owner, operation }));
+      // else security is disabled so let the operation proceed
+    } catch (error) {
+      this.authLogger.genericOperation({ operation, savedObjectID, error });
```

Comment:
I think your original idea of having the Authorization class handle this check avoids some repetition of the `try/catch` logic. We could also just pull it into a function outside of this class. Let me know what you think though.

Reply:
I think is fine as you did it. What are the benefits of a function outside of the class?

Reply:
I can't really think of many benefits I guess the alternative is the function outside of the class would need to take a bunch of parameters because it'd need to invoke the `authorization` and other things to handle the `try/catch` logic. I guess if it gets cumbersome to keep it in the class we can move it out later 🤷‍♂️